src/parsers/transformer.rs

```diff
@@ -62,16 +92,21 @@ pub fn transform<'line>(
         TransformerType::FakeBase16String => Cow::from(fake_base16_string()),
         TransformerType::FakeBase32String => Cow::from(fake_base32_string()),
         TransformerType::FakeCity => Cow::from(CityName().fake::<String>()),
-        TransformerType::FakeCompanyName => Cow::from(fake_company_name(&transformer.args, unique)),
-        TransformerType::FakeEmail => Cow::from(fake_email(&transformer.args, unique)),
+        TransformerType::FakeCompanyName => {
+            Cow::from(fake_company_name(value, &transformer.args, unique))
+        }
+        TransformerType::FakeEmail => Cow::from(fake_email(value, &transformer.args, unique)),
         TransformerType::FakeEmailOrPhone => {
             Cow::from(fake_email_or_phone(value, &transformer.args, unique))
         }
-        TransformerType::FakeFirstName => Cow::from(FirstName().fake::<String>()),
+        TransformerType::FakeFirstName => Cow::from(fake_first_name(value, None)),
+        TransformerType::DeterministicFirstName => Cow::from(fake_first_name(value, id)),
```

Comment:
@sarahfraser Instead of creating a new TransformerType here, we could have teams set the deterministic argument to true in the FakeFirstName transformer and assign the ID. The fake_first_name will then generate either a deterministic or random name based on whether the deterministic flag is set.